noir-projects/noir-contracts/contracts/avm_test_contract/src/main.nr

```diff
@@ -1,38 +1,9 @@
-use dep::aztec::{macros::aztec, protocol_types::traits::{Deserialize, Packable, Serialize}};
-
-// I tried using #[derive(Serialize, Deserialize)] macro here but for whatever reason it fails to compile.
```

Comment:
Moved the note to a separate file because then the derivation works.

I find it quite weird to have the note defined in the same file as the contract.

Do we find this bug worthy of trying to fix?